### ISSUE TITLE
Assert in parallel with message - weird behaviour

### ISSUE THREAD

**Issue by Paulie14** [flow123d/flow123d#950]
Running on 2 processors, I do not understand why
```ASSERT(mesh->get_el_ds()->np() == 1);```
ends up with assert as I expected,
and
```ASSERT(mesh->get_el_ds()->np() == 1).error("XFEM supported only on single processor.\n");```
ends up with PETSC error on invalid pointer.
I wrote this assert in ```MHDofHandler::reinit()```.
Why? Is this intended behaviour?

**Comment by Paulie14** [comment]
Reproduced in simple commit in branch PE_assert_bug.

**Comment by dflanderka** [comment]
Bug solved in branch PE_assert_bug, GCC sanitizer moved to new issue.
